### PR TITLE
feat: [CI-6220]: Added parallel/step-group stage/step support for CI and dockerInfra support

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Added parallel/step-group stage/step support for CI and dockerInfra support
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40568)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java [hunk 1]
```diff
@@ -751,22 +755,25 @@ private static Map<String, String> getSslVerifyEnvVars(ParameterField<Boolean> s
...
+  private Map<String, String> getBuildEnvVars(
+      Ambiance ambiance, Pair<String, String> buildEnvVar, ConnectorDetails gitConnector) {
     Map<String, String> map = new HashMap<>();
     if (buildEnvVar != null) {
-      String type = buildEnvVar.getKey();
-      setMandatoryEnvironmentVariable(map, type, buildEnvVar.getValue());
-      if (DRONE_TAG.equals(type)) {
-        setMandatoryEnvironmentVariable(map, DRONE_BUILD_EVENT, TAG_BUILD_EVENT);
+      switch (buildEnvVar.getKey()) {
+        case DRONE_TAG:
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/PluginSettingUtils.java:768]: Why Drone ?

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/steps/CIStepsPlanCreator.java [hunk 2]
```diff
@@ -81,14 +80,9 @@ public PlanNode createPlanForParentNode(PlanCreationContext ctx, YamlField confi
...
     LinkedHashMap<String, PlanCreationResponse> responseMap = new LinkedHashMap<>();
-    List<YamlField> stages = getStepYamlFields(config);
-    List<ExecutionWrapperConfig> executionWrapperConfigs = stages.stream()
-                                                               .map(stage
-                                                                   -> ExecutionWrapperConfig.builder()
-                                                                          .uuid(stage.getUuid())
-                                                                          .step(stage.getNode().getCurrJsonNode())
-                                                                          .build())
-                                                               .collect(Collectors.toList());
+    List<YamlField> stages = CIPlanCreatorUtils.getStepYamlFields(config);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/steps/CIStepsPlanCreator.java:83]: steps
add null check

Comment (raghavharness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/steps/CIStepsPlanCreator.java]: changed to steps. Null check is already present in getStepYamlFields func.

**Comment by raghavharness** [comment]
trigger codebasehash

**Comment by raghavharness** [comment]
trigger codebasehash

**Review by harshjain12 — COMMENTED** [review]
> Reviewed 3 of 4 files at r1, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40568)* status: 0 of 1 approvals obtained.

**Review by harshjain12 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40568)* status: :shipit: complete! ✓

**Pull request merged**